homeassistant/components/syncthru/sensor.py

```diff
@@ -173,8 +186,25 @@ async def async_update(self):
                 self.syncthru.url,
             )
             self._active = False
-        self._state = SYNCTHRU_STATE_HUMAN[self.syncthru.device_status()]
-        self._attributes = {"display_text": self.syncthru.device_status_details()}
+        self._state = self.syncthru.is_online()
+        self._attributes = {
+            "detailed_state": SYNCTHRU_STATE_HUMAN[self.syncthru.device_status()],
+            "display_text": self.syncthru.device_status_details(),
+        }
+
+
+class SyncThruProblemSensor(SyncThruSensor):
```

Comment:
If it's a binary sensor it should move to a binary sensor platform.

Reply:
I was unsure whether I could merge the current Syncthru super class with the BinaryEntitty super class and have now defined a binary syncthru super class separately. I hope this fulfils the intention.

Reply:
We need to create a `binary_sensor.py` module where we create the entities and add them etc.

Reply:
Okay, I split the definitions. Do I have to notify the config_flow setup that there are multiple sensor types for this integration or does it find out by itself?

Reply:
We should forward the config entry setup to each platform from the init module `async_setup_entry`.

Reply:
done

Reply:
done